gse-afs-ext-base/src/main/java/com/powsybl/gse/afs/ext/base/ContingencyStoreEditor.java

```diff
@@ -45,29 +46,88 @@
     private final class ContingencyTreeCell extends TreeCell<Object> {
 
         private ContingencyTreeCell() {
-            setOnDragDropped(event -> {
-                Dragboard db = event.getDragboard();
-                boolean success = false;
+            setOnDragDetected(this::onDragDetected);
+            setOnDragOver(this::onDragOver);
+            setOnDragDropped(this::onDragDropped);
+        }
+
+        private TreeItem<Object> checkTreeItem() {
```

Comment:
Find a better name for this method. If I understand well, it get the TreeItem<Contingency> corresponding the selected element or null, if nothing is selected.
Proposal: `getContingencyTreeItem()`

This code is duplicate in addContingencyElement. Refactor addContingencyElement to use your method.